rclpy/rclpy/parameter_client.py

```diff
@@ -309,7 +309,9 @@ def load_parameter_file(
         :param use_wildcard: Whether to use wildcard expansion.
         :return: Future with the result from the set_parameters call.
         """
-        param_dict = parameter_dict_from_yaml_file(parameter_file, use_wildcard)
+        print(self.remote_node_name)
+        param_dict = parameter_dict_from_yaml_file(
+            parameter_file, use_wildcard, target_nodes=[self.remote_node_name])
```

Comment:
`remote_node_name` should be specified so that it can know what needs to be parsed from parameter yaml file.